python/lsst/ctrl/ingestd/config.py

```diff
@@ -19,109 +19,49 @@
-
-            if not rucio_prefix.endswith("/"):
-                rucio_prefix = rucio_prefix + "/"
-                self._topic_dict[topic]["rucio_prefix"] = rucio_prefix
-
-            fs_prefix = mapping_dict.get("fs_prefix", None)
-            if fs_prefix:
-                if not fs_prefix.endswith("/"):
-                    fs_prefix = fs_prefix + "/"
-            else:
-                fs_prefix = ""
-            self._topic_dict[topic]["fs_prefix"] = fs_prefix
-
-    def get_num_messages(self) -> int:
-        """Getter method for number of Kafka messages to process at a time"""
-        return self._num_messages
-
-    def get_timeout(self) -> int:
-        """Getter method for length of time to wait for Kafka messages"""
-        return self._timeout
-
-    def get_topic_dict(self) -> dict:
-        """Getter method for topic to local prefix mapping"""
-        return self._topic_dict
-
-    def get_topics(self) -> list:
-        """Getter method for Kafka topics"""
-        return list(self._topic_dict.keys())
-
-    def get_brokers(self) -> str:
-        """Getter method for Kafka brokers"""
-        return self._brokers
-
-    def get_client_id(self) -> str:
-        """Getter method for Kafka client.id"""
-        return self._client_id
-
-    def get_group_id(self) -> str:
-        """Getter method for Kafka group.id"""
-        return self._group_id
-
-    def get_butler_repo(self) -> str:
-        """Getter method for the butler repo"""
-        return self._butler_repo
+from pydantic import BaseModel, Field, computed_field, model_validator
+
+
+class _TopicModel(BaseModel):
+    rucio_prefix: str
+    fs_prefix: str = ""
+
+    @model_validator(mode="after")
+    def process_strings(self) -> "_TopicModel":
+        if not self.rucio_prefix.endswith("/"):
+            self.rucio_prefix += "/"
+        if self.fs_prefix and not self.fs_prefix.endswith("/"):
+            self.fs_prefix += "/"
+        return self
+
+
+class Config(BaseModel):
+    brokers: list[str]
+    client_id: str = Field(default_factory=lambda: socket.gethostname())
+    group_id: str
+    num_messages: int = 50
+    timeout: int = 1
+    butler_repo: str
+    topics: dict[str, _TopicModel] = Field(min_length=1)
+
+    @classmethod
+    def load(cls, config_file: str) -> "Config":
+        try:
+            with open(config_file) as file:
+                config_dict = yaml.load(file, Loader=yaml.FullLoader)
+            return cls.model_validate(config_dict)
+        except yaml.YAMLError as e:
+            raise ValueError(f"Error parsing {config_file}: {e}") from e
+        except Exception as e:
+            raise RuntimeError(f"Unexpected error loading {config_file}: {e}") from e
+
+    @computed_field
+    def brokers_as_string(self) -> str:
+        return ",".join(self.brokers)
+
+    @computed_field
+    def topics_as_list(self) -> list:
+        return list(self.topics.keys())
```

Comment:
`list(self.topics)` should work just fine here.

Reply:
Fixed